### PR TITLE
refactor(meta/config): Adapt RFC Config Backward Compatibility

### PR DESCRIPTION
**Pull request by Xuanwo**
Signed-off-by: Xuanwo <github@xuanwo.io>

I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Part of https://github.com/datafuselabs/databend/issues/5297

After this PR, `databend-meta` will maintain config backward compatibility.

## Changelog

- Improvement

### CODE REVIEW

**Comment by Xuanwo** [comment]
@drmingdrmer @lichuang metasrv config is so wired that confused with `metasrv_xxx` and  `kv_xxx`, how about deprecating all `kv_xxx` configs?

File: metasrv/tests/it/configs.rs [hunk 1]
```diff
@@ -95,12 +89,14 @@ sled_tree_prefix = "sled_foo"
     assert_eq!(cfg.raft_config.id, 20);
     assert_eq!(cfg.raft_config.sled_tree_prefix, "sled_foo");
 
+    std::env::remove_var("METASRV_CONFIG_FILE");
+
```
Comment (drmingdrmer) [inline metasrv/tests/it/configs.rs:93]: Won't setting a process-wise env will affect test cases in other threads?

**Comment by Xuanwo** [comment]
While addressing the comment @drmingdrmer proposed, I found other serious bugs to address.

I will convert this PR to draft until I figure it out.

**Comment by Xuanwo** [comment]
In short, serde doesn't work correctly along with `alias`: https://github.com/serde-rs/serde/pull/1519

**Comment by Xuanwo** [comment]
@drmingdrmer Please take a look.

**Review by drmingdrmer — APPROVED** [review]
Nice and neat

**Pull request merged**